core/interop/Streams/ManagedLib/StreamAsCExports.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace InteropWithStream;
+
+/// <summary>
+/// Stream data type projected as an opaque handle and interacted via C exports.
+/// </summary>
+public static unsafe partial class StreamExports
+{
+#if NETFRAMEWORK
+    [DNNE.Export]
+#else
+    [UnmanagedCallersOnly]
+#endif // !NETFRAMEWORK
+    public static IntPtr C_CreateStream()
+    {
+        MemoryStream stream = new();
+        return GCHandle.ToIntPtr(GCHandle.Alloc(stream));
+    }
+
+#if NETFRAMEWORK
+    [DNNE.Export]
+#else
+    [UnmanagedCallersOnly]
+#endif // !NETFRAMEWORK
+    public static void C_DeleteStream(IntPtr streamMaybe)
+    {
+        if (streamMaybe != IntPtr.Zero)
+        {
+            GCHandle.FromIntPtr(streamMaybe).Free();
+        }
+    }
+
+#if NETFRAMEWORK
+    [DNNE.Export]
+#else
+    [UnmanagedCallersOnly]
+#endif // !NETFRAMEWORK
+    public static int C_PrintStream(IntPtr streamMaybe)
```

Comment:
"streamMaybe"?  The code as written will throw and then fail with an HResult if it's not.